### PR TITLE
Email address enrichment fix

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39385

## Description
- The playbook was sending to the `WhereFieldEquals` transformer a list of items where some being of type `dict` and other of type `string`. The transformer should only work on dicts so I filtered only the content to get only the dicts with the relevant field.
- That being said the transformer should not "crash" on such errors and should just ignore them. Thus Itay added condition to ignore non-dicts in the script.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 1]
```diff
@@ -397,8 +397,13 @@ tasks:
           - operator: uniq
       email:
         complex:
-          root: Account
-          accessor: Email
+          root: Account.Email
```
Comment (idovandijk) [inline Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml:400]: Need to make sure that changing the root accessor is not breaking compatibility for < 6.0 versions

**Review by idovandijk — COMMENTED** [review]
Looks good but please check < 6.0 behavior because this playbook is widely used

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml (hunk 1 above)
Comment (ShahafBenYakir) [reply Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml:400]: Would it @Itay4 ?

**Comment by ShahafBenYakir** [comment]
Failing test for WhereKeyEquals fix:
https://app.circleci.com/pipelines/github/demisto/content/107559/workflows/e60df7a8-1108-4e7d-ac58-ee4a4ccf8c9e/jobs/411730

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml (hunk 1 above)
Comment (Itay4) [reply Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: probably better to keep it as it was

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 2]
```diff
@@ -397,8 +397,13 @@ tasks:
...
+          filters:
+          - - operator: isNotEmpty
```
Comment (Itay4) [inline Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml:402]: should we have 2 dashes here ?

**Comment by ShahafBenYakir** [comment]
@idovandijk look now

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml (hunk 2 above)
Comment (ShahafBenYakir) [reply Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml:402]: Yes see other occurrences in the file. For [example](https://github.com/demisto/content/blob/8cc72055c3ce5daab06074d10691e30a16c4946f/Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml#L107)

File: Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml (hunk 1 above)
Comment (ShahafBenYakir) [reply Packs/CommonPlaybooks/Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: Done ✔️

**Review by idovandijk — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
Tested locally since we have an issue with the instance

**Pull request merged**